core/src/main/java/org/apache/calcite/sql/SqlRename.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+
+/**
+ * Base class for {@code RENAME} statement parse tree nodes.
+ */
+public abstract class SqlRename extends SqlDdl {
+
+  public enum RenameOption {
+    /**
+     * Query used the TO keyword.
+     */
+    TO,
+
+    /**
+     * Query used the AS keyword.
+     */
+    AS
+  }
+
+  private final SqlIdentifier targetStructure;
+  private final SqlIdentifier sourceStructure;
+  private final RenameOption renameOption;
+
+  /**
+   * Creates a {@code SqlRename}.
+   * @param operator   The specific RENAME operator
+   * @param pos  Parser position, must not be null
+   * @param targetStructure  The structure being renamed
+   * @param sourceStructure  What the structure is being renamed to
+   * @param renameOption  Whether the query was specified using TO or AS
+   */
+  protected SqlRename(SqlOperator operator, SqlParserPos pos,
+      SqlIdentifier targetStructure, SqlIdentifier sourceStructure,
+      RenameOption renameOption) {
+    super(operator, pos);
+    this.targetStructure = targetStructure;
+    this.sourceStructure = sourceStructure;
+    this.renameOption = renameOption;
```

Comment:
To simplify things a little bit, do we even need to track the rename option? If there's no behavioral differences between the two cases, can we always just unparse TO and AS to the same value (e.g. AS)?

Reply:
Not sure, I kept it his way since that is how it is done by SqlRenameTable. It would probably work to just unparse to AS each time. If so I can remove the enum completely and implement this change.

Reply:
If it works the same, then I'd say let's go with the simpler approach of not having the enum at all.